recipes/meson-python/meta.yaml

```diff
@@ -0,0 +1,63 @@
+{% set name = "meson-python" %}
+{% set version = "0.4.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/meson_python-{{ version }}.tar.gz
+  sha256: 841d68a9ff26b7f75c8aab23008966abc52646846e6f1b50de55f8961834ed0f
+  patches:
+    - ninja-overdep.patch
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - colorama
+    - meson >=0.60.0
+    - ninja
+    - pep621 >=0.3.0
+    - pip
+    - python >=3.7
+    - tomli >=1.0.0
+    - typing-extensions >=3.7.4
```

Comment:
`patchelf` is also needed, see `mesonpy/_elf.py`.

Reply:
On Linux only

Reply:
Since we are building for noarch now tests will only run on Linux.

Reply:
There's actually a good argument for leaving it out, but I'm not sure if it's currently optional. It will depend on whether there's a standalone non-Python library being vendored.

Actually, on second though - maybe fine to leave it out until someone needs it. WDYT?

Reply:
We cannot depend on platform specific libraries with a noarch package, if platform specific dependencies are required, this package cannot be noarch.

Reply:
Ah, then let's leave it out - SciPy doesn't need it, and I don't think most other packages need it either.

Reply:
> We cannot depend on platform specific libraries with a noarch package, if platform specific dependencies are required, this package cannot be noarch.

That's [technically not true](https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python). If you have a dependency that is available only available for some platforms, we still recommend noarch it's just that the package will be uninstallable for platforms which do not have the dependency.

There are two ways to remedy this:

1. If the dependency is required, accept that the package is uninstallable on some platforms because it doesn't work on those platforms but the package itself is pure python.
2. If the dependency is optional, you can remove the package from the run to run_constrained and users need to manually install if they want. OR ask upstream to add an empty metapackage for the platforms that are missing the package.

Reply:
Well, that's not what I meant, but point taken ;). I'll switch to noarch and drop patchelf from the run requirements.